### PR TITLE
Add support for GRPC API to H2O-3

### PR DESCRIPTION
**Pull request by st-pasha**
Related issues: [PUBDEV-2370](https://0xdata.atlassian.net/browse/PUBDEV-2370), [PUBDEV-3650](https://0xdata.atlassian.net/browse/PUBDEV-3650)

### CODE REVIEW

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by mmalohlava** [comment]
The project import works in my idea. 

I realized one issue: putting grpc module into assembly is going to increase `h2o.jar` size (bringing problems with PyPi).

**Comment by navdeep-G** [comment]
@mmalohlava @st-pasha Tiny note: We just get the jar size small enough for PyPi. Currently, we are ~3MB under the limit for PyPi.

**Comment by st-pasha** [comment]
We don't have to put it into the "general" assembly, just into the one that is going to be shipped with `py2o`. And for that assembly we can exclude many other features such as hadoop / parquet / orc / flow / rest / etc.

**Comment by mmalohlava** [comment]
Ok, sounds good for me - let's remove it from main `h2o.jar` assembly for now.

File: settings.gradle [hunk 1]
```diff
@@ -6,7 +6,8 @@ include 'h2o-web'
 include 'h2o-app'
 include 'h2o-r'
 include 'h2o-py'
-include 'h2o-assembly'
+include 'h2o-assemblies:main'
```
Comment (mmalohlava) [inline settings.gradle]: interesting - does it work?

Comment (st-pasha) [inline settings.gradle]: as far as I can see, it does. The gradle build completes successfully; builds both versions of h2o.jar; and both jars can start without throwing any exceptions

**Comment by st-pasha** [comment]
Removed GRPC from "main" h2o assembly; added it to the "py2o" assembly. Also for the latter excluded hdfs / parquet / flow / avro support -- it reduces the size of the jar file down to 18Mb only.

File: h2o-app/build.gradle [hunk 2]
```diff
@@ -4,12 +4,10 @@
 description = "H2O Application Runner"
 
 dependencies {
-  compile project(":h2o-web")
```
Comment (mmalohlava) [inline h2o-app/build.gradle]: This is breaking change!! :-(

**Pull request merged**